packages/codec/lib/core.ts

```diff
@@ -850,7 +851,7 @@ function* decodeBytecode(
         value = yield* decode(dataType, variable.pointer, info, {
           allowRetry: true, //we know we're in full mode
           strictAbiMode: true,
-          paddingMode: "zero" //force zero-padding!
+          paddingMode: "defaultOrZero"
```

Comment:
Why is this necessary?

Reply:
So, this code is for decoding immutables.  Prior to Solidity 0.8.9, immutables, rather than using the normal padding, were always zero-padded regardless of type.  However, starting in 0.8.9, they use normal padding.  So, previously this code was set to assume zero-padding.  However, now I've changed it to tolerate either zero-padding or normal padding.  Notionally, I could have it specifically check for one or the other based on the compiler version, but I really prefer to avoid version checks wherever possible, so instead I just had it tolerate either.

Reply:
Ah, that context helps! Thank you!